src/Analyser/NodeScopeResolver.php

```diff
@@ -1956,8 +1956,20 @@ function (MutatingScope $scope) use ($expr, $nodeCallback, $context): Expression
 						$nameType->getCallableParametersAcceptors($scope),
 					);
 				}
+
 				$nameResult = $this->processExprNode($expr->name, $scope, $nodeCallback, $context->enterDeep());
-				$throwPoints = $nameResult->getThrowPoints();
+				if ($nameType->isObject()->yes()) {
+					$invokeResult = $this->processExprNode(
+						new MethodCall($expr->name, '__invoke', $expr->getArgs(), $expr->getAttributes()),
+						$scope,
+						static function (): void {
+						},
+						$context->enterDeep(),
+					);
+					$throwPoints = $invokeResult->getThrowPoints();
```

Comment:
We don't want to ignore `$nameResult` throw points even in this branch.

Reply:
I used array_merge instead